tests/serverpod_test_server/lib/src/endpoints/test_tools.dart

```diff
@@ -185,6 +185,25 @@ class TestToolsEndpoint extends Endpoint {
     session.log('test session log in endpoint');
   }
 
+  static bool willCloseListenerCalled = false;
```

Comment:
Another concern might be concurrency. Currently this method is called within the same file, what if another file calls this endpoint?

Reply:
Great suggestion! I will change to completer. If another test needs the same thing they will simply have to create their own endpoint. At least now it will throw and it will be obvious 💯

Reply:
40dcfff

Reply:
What if another file calls this endpoint after the "setup" for the first file is completed? Perhaps the completer can be a map of some sort? Then concurrency won't be an issue.

Reply:
Wait, I think this will be isolated within each `withServerpod` anyway. So it might not be a problem as long as we start a separate server in each test file.

Reply:
> Wait, I think this will be isolated within each withServerpod anyway. So it might not be a problem as long as we start a separate server in each test file.

Yes, that should be the case! Each `withServerpod` instantiates their own Endpoints objects